kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchHTTPManager.java

```diff
@@ -0,0 +1,278 @@
+      ret.setDaemon(true);
+      return ret;
+    }
+  });
+
+  public WatchHTTPManager(final OkHttpClient client,
+    final BaseOperation<T, L, ?, ?> baseOperation,
+    final String version, final Watcher<T> watcher, final int reconnectInterval,
+    final int reconnectLimit, long connectTimeout)
+    throws MalformedURLException {
+
+    if (version == null) {
+      L currentList = baseOperation.list();
+      this.resourceVersion = new AtomicReference<>(currentList.getMetadata().getResourceVersion());
+    } else {
+      this.resourceVersion = new AtomicReference<>(version);
+    }
+    this.baseOperation = baseOperation;
+    this.watcher = watcher;
+    this.reconnectInterval = reconnectInterval;
+    this.reconnectLimit = reconnectLimit;
+
+    OkHttpClient clonedClient = client.newBuilder()
+      .connectTimeout(connectTimeout, TimeUnit.MILLISECONDS)
+      .readTimeout(0,TimeUnit.MILLISECONDS)
+      .cache(null)
+      .build();
+
+    // If we set the HttpLoggingInterceptor's logging level to Body (as it is by default), it does
+    // not let us stream responses from the server.
+    for (Interceptor i : clonedClient.networkInterceptors()) {
+      if (i instanceof HttpLoggingInterceptor) {
+        HttpLoggingInterceptor interceptor = (HttpLoggingInterceptor) i;
+        interceptor.setLevel(HttpLoggingInterceptor.Level.BASIC);
+      }
+    }
+
+    this.clonedClient = clonedClient;
+    requestUrl = baseOperation.getNamespacedUrl();
+    scheduleReconnect();
+  }
+
+  private final void runWatch() {
+    logger.debug("Watching via HTTP GET ... {}", this);
+
+    HttpUrl.Builder httpUrlBuilder = HttpUrl.get(requestUrl).newBuilder();
+    String labelQueryParam = baseOperation.getLabelQueryParam();
+    if (isNotNullOrEmpty(labelQueryParam)) {
+      httpUrlBuilder.addQueryParameter("labelSelector", labelQueryParam);
+    }
+
+    String fieldQueryString = baseOperation.getFieldQueryParam();
+    String name = baseOperation.getName();
+    if (name != null && name.length() > 0) {
+      if (fieldQueryString.length() > 0) {
+        fieldQueryString += ",";
+      }
+      fieldQueryString += "metadata.name=" + name;
+    }
+
+    if (isNotNullOrEmpty(fieldQueryString)) {
+      httpUrlBuilder.addQueryParameter("fieldSelector", fieldQueryString);
+    }
+
+    httpUrlBuilder
+      .addQueryParameter("resourceVersion", this.resourceVersion.get())
+      .addQueryParameter("watch", "true");
+
+    final Request request = new Request.Builder()
+      .get()
+      .url(httpUrlBuilder.build())
+      .addHeader("Origin", requestUrl.getProtocol() + "://" + requestUrl.getHost() + ":" + requestUrl.getPort())
+      .build();
+
+    Response response = null;
+    try {
+      response = clonedClient.newCall(request).execute();
+      BufferedSource source = response.body().source();
+      while (!source.exhausted()) {
+        String message = source.readUtf8LineStrict();
```

Comment:
Is each event new line separated then? Not chunked response? I haven't looked to be sure but will trust you :)